x/ecocredit/server/testsuite/tx.go

```diff
@@ -0,0 +1,587 @@
+		expErr  string
+		wantErr bool
+	}{
+		{
+			name:  "invalid sell order",
+			owner: addr1,
+			updates: []*ecocredit.MsgUpdateSellOrders_Update{
+				{
+					SellOrderId:       99,
+					NewQuantity:       "1.0",
+					NewAskPrice:       &askPrice1,
+					DisableAutoRetire: true,
+				},
+				{
+					SellOrderId:       100,
+					NewQuantity:       "1.0",
+					NewAskPrice:       &askPrice1,
+					DisableAutoRetire: true,
+				},
+			},
+			expErr:  "invalid sell order",
+			wantErr: true,
+		},
+		{
+			name:  "unauthorized",
+			owner: addr2,
+			updates: []*ecocredit.MsgUpdateSellOrders_Update{
+				{
+					SellOrderId:       sellRes.SellOrderIds[0],
+					NewQuantity:       "1.0",
+					NewAskPrice:       &askPrice1,
+					DisableAutoRetire: true,
+				},
+				{
+					SellOrderId:       sellRes.SellOrderIds[1],
+					NewQuantity:       "1.0",
+					NewAskPrice:       &askPrice1,
+					DisableAutoRetire: true,
+				},
+			},
+			expErr:  "unauthorized",
+			wantErr: true,
+		},
+		{
+			name:  "insufficient credit balance",
+			owner: addr1,
+			updates: []*ecocredit.MsgUpdateSellOrders_Update{
+				{
+					SellOrderId:       sellRes.SellOrderIds[0],
+					NewQuantity:       "99",
+					NewAskPrice:       &askPrice1,
+					DisableAutoRetire: true,
+				},
+				{
+					SellOrderId:       sellRes.SellOrderIds[1],
+					NewQuantity:       "99",
+					NewAskPrice:       &askPrice1,
+					DisableAutoRetire: true,
+				},
+			},
+			expErr:  "insufficient credit balance",
+			wantErr: true,
+		},
+		// TODO: Verify that NewAskPrice.Denom is in AllowAskDenom #624
+		//{
+		//	name: "denom not allowed",
+		//	owner: addr1,
+		//	updates: []*ecocredit.MsgUpdateSellOrders_Update{
+		//		{
+		//			SellOrderId:       sellRes.SellOrderIds[0],
+		//			NewQuantity:       "1.0",
+		//			NewAskPrice:       &askPrice2,
+		//			DisableAutoRetire: true,
+		//		},
+		//		{
+		//			SellOrderId:       sellRes.SellOrderIds[1],
+		//			NewQuantity:       "1.0",
+		//			NewAskPrice:       &askPrice2,
+		//			DisableAutoRetire: true,
+		//		},
+		//	},
+		//	expErr: "denom not allowed",
+		//	wantErr: true,
+		//},
+		{
+			name:  "valid request",
+			owner: addr1,
+			updates: []*ecocredit.MsgUpdateSellOrders_Update{
+				{
+					SellOrderId:       sellRes.SellOrderIds[0],
+					NewQuantity:       "1.0",
+					NewAskPrice:       &askPrice1,
+					DisableAutoRetire: true,
+				},
+				{
+					SellOrderId:       sellRes.SellOrderIds[1],
+					NewQuantity:       "1.0",
+					NewAskPrice:       &askPrice1,
+					DisableAutoRetire: true,
+				},
+			},
+			expErr:  "",
+			wantErr: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			require := s.Require()
+
+			_, err := s.msgClient.UpdateSellOrders(s.ctx, &ecocredit.MsgUpdateSellOrders{
+				Owner:   tc.owner,
+				Updates: tc.updates,
+			})
+
+			if tc.wantErr {
+				require.Error(err)
+				require.Contains(err.Error(), tc.expErr)
+			} else {
+				require.NoError(err)
```

Comment:
should we check that the updates applied to the sell order here?

Reply:
added in https://github.com/regen-network/regen-ledger/pull/626/commits/dc6d6273a6a4dfb67f85ec6581e1b434aee1b500